server/container/guice/blob/strategy/src/main/java/org/apache/james/modules/blobstore/BlobStrategyModule.java

```diff
@@ -0,0 +1,105 @@
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.modules.blobstore;
+
+import java.time.Clock;
+import java.util.Set;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobReferenceSource;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.api.BlobStoreDAO;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.api.MetricableBlobStore;
+import org.apache.james.mailbox.cassandra.mail.AttachmentBlobReferenceSource;
+import org.apache.james.mailbox.cassandra.mail.MessageBlobReferenceSource;
+import org.apache.james.mailrepository.cassandra.MailRepositoryBlobReferenceSource;
+import org.apache.james.queue.rabbitmq.view.cassandra.MailQueueViewBlobReferenceSource;
+import org.apache.james.server.blob.deduplication.BlobGCTaskAdditionalInformationDTO;
+import org.apache.james.server.blob.deduplication.BlobGCTaskDTO;
+import org.apache.james.server.blob.deduplication.GenerationAwareBlobId;
+import org.apache.james.server.task.json.dto.AdditionalInformationDTO;
+import org.apache.james.server.task.json.dto.AdditionalInformationDTOModule;
+import org.apache.james.server.task.json.dto.TaskDTO;
+import org.apache.james.server.task.json.dto.TaskDTOModule;
+import org.apache.james.task.Task;
+import org.apache.james.task.TaskExecutionDetails;
+import org.apache.james.webadmin.dto.DTOModuleInjections;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.multibindings.ProvidesIntoSet;
+import com.google.inject.name.Named;
+
+public class BlobStrategyModule extends AbstractModule {
+
+    @Override
+    protected void configure() {
+        Multibinder<BlobReferenceSource> multiBinder = Multibinder.newSetBinder(binder(),
+            BlobReferenceSource.class);
+
+        multiBinder.addBinding().to(AttachmentBlobReferenceSource.class);
+        multiBinder.addBinding().to(MailQueueViewBlobReferenceSource.class);
+        multiBinder.addBinding().to(MailRepositoryBlobReferenceSource.class);
+        multiBinder.addBinding().to(MessageBlobReferenceSource.class);
```

Comment:
The idea behind the multi-binder is that you don't need the registration to be in a single place.

Please move those in `RabbitMQModule` `CassandraMailRepositoryModule` `CassandraMailboxModule`...